### PR TITLE
Adding extra CI regression tests for scalar advection/diffusion

### PR DESCRIPTION
**Pull request by w3168**
Move scalar_advection.py to tests

add regression tests (and run as part of CI!) for:
scalar_advection.py
scalar_advection_diffusion.py
scalar_advection_diffusion_DH219_skew.py

### CODE REVIEW

File: tests/scalar_advection/test_scalar_advection.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import numpy as np
+
+base = Path(__file__).parent.resolve()
+
+
+def test_scalar_advection():
+    expected_error = np.load(base / "expected_error.npy")
+    final_error = np.loadtxt(base / "final_error.log")
+    print(expected_error)
+    print(final_error)
```
Comment (angus-g) [inline tests/scalar_advection/test_scalar_advection.py:12]:
```suggestion
```

File: tests/scalar_advection_diffusion/generate_expected.py [hunk 2]
```diff
@@ -4,10 +4,18 @@
...
     results[param_str] = errors
 
 np.savez("expected_errors.npz", **results)
+
+# cases for regression testing
+int_q = np.loadtxt("integrated_q.log")
+np.save("expected_integrated_q.npy", [int_q])
+
+int_q_DH219 = np.loadtxt("integrated_q_DH219.log")
+np.save("expected_integrated_q_DH219.npy", [int_q_DH219])
```
Comment (angus-g) [inline tests/scalar_advection_diffusion/generate_expected.py:21]: These don't get tested anywhere?

**Review by angus-g — CHANGES_REQUESTED** [review]
Yay, more tests!

File: tests/scalar_advection_diffusion/generate_expected.py (hunk 2 above)
Comment (w3168) [reply tests/scalar_advection_diffusion/generate_expected.py:21]: I forgot to add the test file! I've added it now

**Review by angus-g — APPROVED** [review]

**Pull request merged**